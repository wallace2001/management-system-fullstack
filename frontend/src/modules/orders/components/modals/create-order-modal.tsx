'use client';

import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
} from '@/components/ui/dialog';
import { Button } from '@/components/ui/button';
import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import { Form, FormField, FormItem, FormLabel, FormControl, FormMessage } from '@/components/ui/form';
import { useCreateOrder } from '../../hooks/use-order';
import { useOrdersModal } from '../../store/use-orders-modals-store';
import { OrderFormValues, orderSchema } from '../../schemas/order-schema';
import { ProductMultiSelect } from '@/components/product-multi-select';

export function CreateOrderModal() {
  const { createOpen, close } = useOrdersModal();
  const { mutate, isPending } = useCreateOrder();

  const form = useForm<OrderFormValues>({
    resolver: zodResolver(orderSchema),
    defaultValues: {
      products: {},
      status: 'PENDING',
    },
  });

  const onSubmit = (values: OrderFormValues) => {
    console.log(values);
    mutate(values, {
      onSuccess: () => {
        form.reset();
        close();
      },
    });
  };

  return (
    <Dialog open={createOpen} onOpenChange={close}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Novo Pedido</DialogTitle>
        </DialogHeader>

        <Form {...form}>
          <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
            <FormField
              control={form.control}
              name="products"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Produtos</FormLabel>
                  <FormControl>
                    <ProductMultiSelect {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <Button className="w-full" disabled={isPending}>
              Criar Pedido
            </Button>
          </form>
        </Form>
      </DialogContent>
    </Dialog>
  );
}
