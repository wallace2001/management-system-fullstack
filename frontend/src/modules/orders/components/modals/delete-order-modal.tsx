'use client';

import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
} from '@/components/ui/dialog';
import { Button } from '@/components/ui/button';
import { useOrdersModal } from '../../store/use-orders-modals-store';
import { useDeleteOrder } from '../../hooks/use-order';
import { handleError } from '@/modules/errors/request-error';

export function DeleteOrderModal() {
  const { deleteOpen, selected, close } = useOrdersModal();
  const { mutate, isPending } = useDeleteOrder();

  const onDelete = () => {
    if (!selected?.id) return;
    mutate(selected.id, {
      onSuccess: () => close(),
      onError: handleError,
    });
  };

  if (!selected) return null;

  return (
    <Dialog open={deleteOpen} onOpenChange={close}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Cancelar pedido ?</DialogTitle>
        </DialogHeader>
        <DialogDescription />

        <p>Tem certeza que deseja cancelar este pedido?</p>

        <div className="flex justify-end gap-2 pt-4">
          <Button variant="outline" onClick={close}>
            Sair
          </Button>
          <Button variant="destructive" onClick={onDelete} disabled={isPending}>
            Cancelar
          </Button>
        </div>
      </DialogContent>
    </Dialog>
  );
}
