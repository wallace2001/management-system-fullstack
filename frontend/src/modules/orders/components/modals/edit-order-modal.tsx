'use client';

import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
} from '@/components/ui/dialog';
import { Button } from '@/components/ui/button';
import { useForm, Controller } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import {
  Form,
  FormItem,
  FormLabel,
  FormControl,
  FormMessage,
} from '@/components/ui/form';
import { useOrdersModal } from '../../store/use-orders-modals-store';
import { useUpdateOrder } from '../../hooks/use-order';
import {
  OrderFormValues,
  orderSchema,
} from '../../schemas/order-schema';
import { ProductMultiSelect } from '@/components/product-multi-select';

export function EditOrderModal() {
  const { editOpen, selected, close } = useOrdersModal();
  const { mutate, isPending } = useUpdateOrder();

  const form = useForm<OrderFormValues>({
    resolver: zodResolver(orderSchema),
    values: {
      products:
        selected?.products && Array.isArray(selected.products)
          ? selected.products.reduce((acc, item) => {
              acc[item.productId] = item.quantity;
              return acc;
            }, {} as Record<string, number>)
          : {},
      status: selected?.status || 'PENDING',
    },
  });

  const onSubmit = (values: OrderFormValues) => {
    if (!selected?.id) return;
    mutate(
      {
          id: selected.id,
          status: values.status,
          products: values.products,
      },
      {
        onSuccess: () => close(),
      }
    );
  };

  if (!selected) return null;

  return (
    <Dialog open={editOpen} onOpenChange={close}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Editar Pedido</DialogTitle>
        </DialogHeader>

        <Form {...form}>
          <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
            <Controller
              name="products"
              control={form.control}
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Produtos</FormLabel>
                  <FormControl>
                    <ProductMultiSelect
                      value={field.value}
                      onChange={field.onChange}
                      className="w-full"
                    />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <Button type="submit" className="w-full" disabled={isPending}>
              Salvar Alterações
            </Button>
          </form>
        </Form>
      </DialogContent>
    </Dialog>
  );
}
