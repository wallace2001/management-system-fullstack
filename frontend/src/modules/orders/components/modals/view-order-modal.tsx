'use client';

import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
} from '@/components/ui/dialog';
import { useOrdersModal } from '../../store/use-orders-modals-store';
import { ScrollArea } from '@/components/ui/scroll-area';
import { Badge } from '@/components/ui/badge';

export function ViewOrderModal() {
  const { detailOpen, selected, close } = useOrdersModal();

  if (!selected) return null;

  return (
    <Dialog open={detailOpen} onOpenChange={close}>
      <DialogContent className="max-w-lg">
        <DialogHeader>
          <DialogTitle>Detalhes do Pedido</DialogTitle>
        </DialogHeader>

        <div className="space-y-4">
          <div>
            <span className="font-semibold">ID:</span> {selected.id}
          </div>
          <div>
            <span className="font-semibold">Status:</span>{' '}
            <Badge variant={
              selected.status === 'COMPLETED' ? 'default' :
              selected.status === 'PENDING' ? 'secondary' : 'destructive'
            }>
              {selected.status}
            </Badge>
          </div>
          <div>
            <span className="font-semibold">Total:</span> R$ {selected.total.toFixed(2)}
          </div>
          <div>
            <span className="font-semibold">Data:</span>{' '}
            {new Date(selected.createdAt).toLocaleDateString()}
          </div>

          <div>
            <span className="font-semibold block mb-2">Produtos:</span>
            <ScrollArea className="h-48 pr-2">
              <ul className="space-y-2">
                {selected.products.map((item) => (
                  <li key={item.productId} className="text-sm">
                    <div className="flex justify-between">
                      <span className="font-medium">{item.product.name}</span>
                      <span>
                        {item.quantity}x - R$ {item.product.price.toFixed(2)}
                      </span>
                    </div>
                  </li>
                ))}
              </ul>
            </ScrollArea>
          </div>
        </div>
      </DialogContent>
    </Dialog>
  );
}