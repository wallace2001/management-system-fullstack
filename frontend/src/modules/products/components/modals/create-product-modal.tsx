'use client';

import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
} from '@/components/ui/dialog';
import { Button } from '@/components/ui/button';
import { Input } from '@/components/ui/input';
import {
  Form,
  FormItem,
  FormLabel,
  FormControl,
  FormField,
  FormMessage,
} from '@/components/ui/form';
import { useForm } from 'react-hook-form';
import { z } from 'zod';
import { useProductModals } from '../../store/use-products-modal-store';
import { useCreateProduct } from '../../hooks/use-product';
import { zodResolver } from '@hookform/resolvers/zod';
import { CurrencyInput } from '@/components/currency-input';

const createProductSchema = z.object({
  name: z.string().min(1, 'Nome é obrigatório'),
  category: z.string().min(1, 'Categoria é obrigatória'),
  description: z.string().min(1, 'Descrição é obrigatória'),
  price: z.coerce.number().min(0, 'Preço deve ser maior ou igual a 0'),
  stockQuantity: z.coerce.number().min(0, 'Estoque deve ser maior ou igual a 0'),
});

type CreateProductValues = z.infer<typeof createProductSchema>;

export function CreateProductModal() {
  const { createOpen, close } = useProductModals();
  const { mutate, isPending } = useCreateProduct();

  const form = useForm<CreateProductValues>({
    resolver: zodResolver(createProductSchema),
    defaultValues: {
      name: '',
      category: '',
      description: '',
      price: 0,
      stockQuantity: 0,
    },
  });

  function onSubmit(data: CreateProductValues) {
    mutate(data, {
      onSuccess: () => {
        form.reset();
        close();
      },
    });
  }

  return (
    <Dialog open={createOpen} onOpenChange={close}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Novo Produto</DialogTitle>
        </DialogHeader>

        <Form {...form}>
          <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
            <FormField
              control={form.control}
              name="name"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Nome</FormLabel>
                  <FormControl>
                    <Input placeholder="Nome do produto" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="category"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Categoria</FormLabel>
                  <FormControl>
                    <Input placeholder="Categoria" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="description"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Descrição</FormLabel>
                  <FormControl>
                    <Input placeholder="Descrição do produto" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="price"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Preço</FormLabel>
                  <FormControl>
                    <CurrencyInput placeholder="Preço" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <FormField
              control={form.control}
              name="stockQuantity"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Estoque</FormLabel>
                  <FormControl>
                    <Input type="number" placeholder="Quantidade em estoque" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <Button type="submit" className="w-full" disabled={isPending}>
              Criar Produto
            </Button>
          </form>
        </Form>
      </DialogContent>
    </Dialog>
  );
}
