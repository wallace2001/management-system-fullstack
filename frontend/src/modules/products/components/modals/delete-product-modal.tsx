import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
} from '@/components/ui/dialog';
import { Button } from '@/components/ui/button';
import { useProductModals } from '../../store/use-products-modal-store';
import { useDeleteProduct } from '../../hooks/use-product';
import { handleError } from '@/modules/errors/request-error';

export function DeleteProductModal() {
  const { deleteOpen, selected, close } = useProductModals();
  const { mutate, isPending } = useDeleteProduct();

  if (!selected) return null;

  return (
    <Dialog open={deleteOpen} onOpenChange={close}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Confirmar exclus√£o</DialogTitle>
        </DialogHeader>
        <DialogDescription />
        <p className="text-sm">
          Deseja remover o produto <strong>{selected.name}</strong>?
        </p>

        <div className="flex justify-end gap-2 pt-4">
          <Button variant="outline" onClick={close}>
            Cancelar
          </Button>
          <Button
            variant="destructive"
            disabled={isPending}
            onClick={() =>
              mutate(selected.id, { onSuccess: close, onError: handleError })
            }
          >
            Confirmar
          </Button>
        </div>
      </DialogContent>
    </Dialog>
  );
}
