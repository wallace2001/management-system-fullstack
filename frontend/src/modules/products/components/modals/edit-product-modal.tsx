'use client';

import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { Input } from '@/components/ui/input';
import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
} from '@/components/ui/dialog';
import {
  Form,
  FormItem,
  FormControl,
  FormLabel,
  FormMessage,
  FormField,
} from '@/components/ui/form';
import { Button } from '@/components/ui/button';
import { useUpdateProduct } from '../../hooks/use-product';
import { useProductModals } from '../../store/use-products-modal-store';
import { z } from 'zod';
import { CurrencyInput } from '@/components/currency-input';

const editProductSchema = z.object({
  id: z.string().uuid(),
  name: z.string().min(1, 'Nome é obrigatório'),
  category: z.string().min(1, 'Categoria é obrigatória'),
  description: z.string().min(1, 'Descrição é obrigatória'),
  price: z.coerce.number().min(0, 'Preço deve ser maior ou igual a 0'),
  stockQuantity: z.coerce.number().min(0, 'Estoque deve ser maior ou igual a 0'),
});

type EditProductValues = z.infer<typeof editProductSchema>;

export function EditProductModal() {
  const { editOpen, selected, close } = useProductModals();
  const { mutate, isPending } = useUpdateProduct();

  const form = useForm<EditProductValues>({
    resolver: zodResolver(editProductSchema),
    defaultValues: {
      id: selected?.id ?? '',
      name: selected?.name ?? '',
      category: selected?.category ?? '',
      description: selected?.description ?? '',
      price: selected?.price ?? 0,
      stockQuantity: selected?.stockQuantity ?? 0,
    },
    values: {
      id: selected?.id ?? '',
      name: selected?.name ?? '',
      category: selected?.category ?? '',
      description: selected?.description ?? '',
      price: selected?.price ?? 0,
      stockQuantity: selected?.stockQuantity ?? 0,
    },
  });

  function onSubmit(data: EditProductValues) {
    mutate(data, {
      onSuccess: () => {
        close();
      },
    });
  }

  if (!selected) return null;

  return (
    <Dialog open={editOpen} onOpenChange={close}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Editar Produto</DialogTitle>
        </DialogHeader>

        <Form {...form}>
          <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
            <FormField
              control={form.control}
              name="name"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Nome</FormLabel>
                  <FormControl>
                    <Input placeholder="Nome do produto" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />
            <FormField
              control={form.control}
              name="category"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Categoria</FormLabel>
                  <FormControl>
                    <Input placeholder="Categoria" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />
            <FormField
              control={form.control}
              name="description"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Descrição</FormLabel>
                  <FormControl>
                    <Input placeholder="Descrição do produto" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />
            <FormField
              control={form.control}
              name="price"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Preço</FormLabel>
                  <FormControl>
                    <CurrencyInput placeholder="Preço" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />
            <FormField
              control={form.control}
              name="stockQuantity"
              render={({ field }) => (
                <FormItem>
                  <FormLabel>Estoque</FormLabel>
                  <FormControl>
                    <Input type="number" placeholder="Estoque" {...field} />
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />

            <Button type="submit" className="w-full" disabled={isPending}>
              Salvar alterações
            </Button>
          </form>
        </Form>
      </DialogContent>
    </Dialog>
  );
}
