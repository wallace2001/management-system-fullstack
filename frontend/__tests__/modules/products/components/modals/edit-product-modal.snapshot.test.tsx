import { render } from '@testing-library/react';
import { EditProductModal } from '@/modules/products/components/modals/edit-product-modal';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

jest.mock('@/modules/products/hooks/use-product', () => ({
  useUpdateProduct: () => ({
    mutate: jest.fn(),
    isPending: false,
  }),
}));

jest.mock('@/modules/products/store/use-products-modal-store', () => ({
  useProductModals: () => ({
    editOpen: true,
    selected: {
      id: '1',
      name: 'Notebook',
      category: 'Tech',
      description: 'Gamer',
      price: 3500,
      stockQuantity: 5,
    },
    close: jest.fn(),
  }),
}));

describe('EditProductModal Snapshot', () => {
  it('renderiza corretamente', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <EditProductModal />
      </QueryClientProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
