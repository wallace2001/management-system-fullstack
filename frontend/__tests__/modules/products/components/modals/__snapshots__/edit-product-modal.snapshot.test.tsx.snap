// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProductModal Snapshot renderiza corretamente 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
